solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/FlowReceiverContainer.java

```diff
@@ -133,6 +137,9 @@ public void unbind() {
 				flowReceiverReference.getStaleMessagesFlag().set(true);
 				flowReceiverReference.get().close();
 				unacknowledgedMessageTracker.reset();
+				if (eventHandler != null) {
+					eventHandler.setHealthStatusDown();
+				}
```

Comment:
Do not mark the health as `DOWN` when unbind is called on the `FlowReceiverContainer`.

This will have the unintended side effect of marking the flow as `DOWN` whenever someone NACKs a message (it triggers a flow rebind within the binder).

Reply:
Setting health `DOWN` (might be we need to use `DISCONNECTED` though or some other custom status), is intentional, as it may not work with other products well enough.

As an example, Connector Manager always show the health of underlying connection in Workflows view. At the moment there is a health status for the whole connection, because of lacking this feature, however after releasing this one, it will catch up aggregated status for every for every binding separately.

In case of removing health indicators completely, there will be nothing to set as a status in a view and thus we need to provide current status there as well, which as I said might be custom, like `DOWN`, `DISCONNECTED` or `UNBIND`.

Reply:
After a careful consideration of all possible options we will proceed with deleting Health Indicators on any `unbind` events.